frame/staking/src/pallet/mod.rs

```diff
@@ -1311,18 +1308,18 @@ pub mod pallet {
 			let ledger = Self::ledger(&controller).ok_or(Error::<T>::NotController)?;
 			ensure!(!ledger.unlocking.is_empty(), Error::<T>::NoUnlockChunk);
 
+			let initial_unlocking = ledger.unlocking.len() as u32;
 			let ledger = ledger.rebond(value);
 			// Last check: the new active amount of ledger must be more than ED.
 			ensure!(ledger.active >= T::Currency::minimum_balance(), Error::<T>::InsufficientBond);
 
 			Self::deposit_event(Event::<T>::Bonded(ledger.stash.clone(), value));
 			Self::update_ledger(&controller, &ledger);
-			Ok(Some(
-				35 * WEIGHT_PER_MICROS +
-					50 * WEIGHT_PER_NANOS * (ledger.unlocking.len() as Weight) +
-					T::DbWeight::get().reads_writes(3, 2),
-			)
-			.into())
+
+			let removed_chunks = 1 + // account the case where last iterated chunk is not removed
```

Comment:
better use safe arithmetic

Reply:
and FYI the dependency of this function on unlocking is super small (substrate weight: `.saturating_add((35_000 as Weight).saturating_mul(l as Weight))`, which is very small)